### PR TITLE
adding MacOS build to workflow

### PR DESCRIPTION
**Pull request by pvmm**
- cleaner code
- one OS per file

### CODE REVIEW

**Comment by pvmm** [comment]
Here is how it looks like: https://github.com/pvmm/debugger/releases/tag/nightly

**Comment by MBilderbeek** [comment]
The Actions say:
Invalid workflow file: .github/workflows/build.yaml#L10
error parsing called workflow
".github/workflows/build.yaml"
-> "./.github/workflows/workflow-build.yaml"
: failed to fetch workflow: workflow was not found.

I guess you're still working on it. Let me know when you're done please. I suppose all actions should be running successfully before we want to merge ;-)

**Comment by pvmm** [comment]
No, it's done. GA complains it cannot find a workflow that is no longer valid because it was replaced. I don't know why the old routine is trying to call the new routine, which is weird. But at least if it doesn't work, we can always revert it.

File: .github/workflows/workflow-build-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Build workflow
+
+on:
+  workflow_call:
+
+env:
+  package_origin: gh
```
Comment (MBilderbeek) [inline .github/workflows/workflow-build-windows.yaml:7]: What's that gh?

Comment (pvmm) [reply .github/workflows/workflow-build-windows.yaml:7]: github

**Comment by MBilderbeek** [comment]
Without having me to read all the yml files... can you summarize what was output before your changes and what is now output after your changes?
Especially because of the thing you said about keeping the old file around. I don't really understand what you said about 'the intermediate step'.

**Comment by pvmm** [comment]
I returned the commit to how it was before (first version of this PR).

When you commented about the error message, I tried to break it into two commits in the same PR, keeping the old workflow-build.yaml around because the old job schedule seems to require it, but that didn't work since Github action will not try to build intermediate steps, so it just does the same thing again.

**Comment by MBilderbeek** [comment]
OK, I will merge it, let's see what happens.
